### PR TITLE
138 new resource security certificate

### LINKED ISSUE #138: [New Resource]: /security/certificates
### Description

Support for /security/certificates

There is no modify

### Requested Resource(s) and/or Data Source(s)

netapp-ontap_security_certificates

### Potential Terraform Configuration

_No response_

### References

https://docs.netapp.com/us-en/ontap-restapi/ontap/swagger-ui/index.html#/security/security_certificate_create

### Would you like to implement a fix?

None

### PR DESCRIPTION
**Pull request by csahu9**
[Issue 138](https://github.com/NetApp/terraform-provider-netapp-ontap/issues/138)

[root@scs000344887 security]# TF_ACC=1 go test `go list ./... | grep -e provider` -v -run=TestAccSecurityCertificateResource
=== RUN   TestAccSecurityCertificateResource
--- PASS: TestAccSecurityCertificateResource (6.89s)
PASS
ok      github.com/netapp/terraform-provider-netapp-ontap/internal/provider/security    6.901s
[root@scs000344887 security]#

### CODE REVIEW

**Review by carchi8py — COMMENTED** [review]
@csahu9  is name returned back in the output for 9.6 and 9.7

If not this code won't work for those system as name is required but would be null.

**Comment by csahu9** [comment]
> @csahu9 is name returned back in the output for 9.6 and 9.7
> 
> If not this code won't work for those system as name is required but would be null.

@carchi8py Resource Read has [version constraints](https://github.com/NetApp/terraform-provider-netapp-ontap/blob/138-new-resource-security_certificates/internal/interfaces/security_certificate.go#L75) before adding `name` to query fields, so it won't be returned for Ontap 9.6/9/7
I've updated the error reporting for scenario where one tries to use `name` in Ontap 9.6/9.7
I've now tested the resource in Ontap 9.6 as well and it is working as expected.

**Comment by chuyich** [comment]
There is one case with minimum setup and the creation has issues:
```
resource "netapp-ontap_security_certificate" "create_certificate" {
  cx_profile_name = "cluster5"
  common_name     = "tfsvm_ca_cert_test"
  type            = "root_ca"
}
```

But creating a certificate without svm info is allowed.

```
csahu-vsim34::security certificate> create -common-name "ttt" -type root-ca -size 2048

The certificate's generated name for reference: ttt_17FCEF1EB6BDE792_ttt

csahu-vsim34::security certificate> show -cert-name ttt_17FCEF1EB6BDE792_ttt
Vserver    Serial Number   Certificate Name                       Type
---------- --------------- -------------------------------------- ------------
csahu-vsim34 
           17FCEF1EB6BDE792 
                           ttt_17FCEF1EB6BDE792_ttt               root-ca
    Certificate Authority: ttt
          Expiration Date: Thu Oct 09 20:14:32 2025
```

So svm_name is optional and computed in the schema definition.

Please take a look and adjust the code for taking care of this case.

**Comment by csahu9** [comment]
> There is one case with minimum setup and the creation has issues:
> 
> ```
> resource "netapp-ontap_security_certificate" "create_certificate" {
>   cx_profile_name = "cluster5"
>   common_name     = "tfsvm_ca_cert_test"
>   type            = "root_ca"
> }
> ```
> 
> But creating a certificate without svm info is allowed.
> 
> ```
> csahu-vsim34::security certificate> create -common-name "ttt" -type root-ca -size 2048
> 
> The certificate's generated name for reference: ttt_17FCEF1EB6BDE792_ttt
> 
> csahu-vsim34::security certificate> show -cert-name ttt_17FCEF1EB6BDE792_ttt
> Vserver    Serial Number   Certificate Name                       Type
> ---------- --------------- -------------------------------------- ------------
> csahu-vsim34 
>            17FCEF1EB6BDE792 
>                            ttt_17FCEF1EB6BDE792_ttt               root-ca
>     Certificate Authority: ttt
>           Expiration Date: Thu Oct 09 20:14:32 2025
> ```
> 
> So svm_name is optional and computed in the schema definition.
> 
> Please take a look and adjust the code for taking care of this case.

@chuyich thanks for pointing this out; I've updated the resource accordingly.

**Review by suhasbshekar — APPROVED** [review]

File: docs/resources/security_certificate.md [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+### Related ONTAP commands
+```commandline
+* security certificate show
+* security certificate create
+* security certificate install
+* security certificate sign
+```
+
+## Supported Platforms
+* On-prem ONTAP system 9.6 or higher
```
Comment (chuyich) [inline docs/resources/security_certificate.md:22]: Just want to make sure if you check the fsx is supported or not.

Comment (csahu9) [reply docs/resources/security_certificate.md:22]: No, @chuyich , I haven't tested for fsx support.
Could you please let me know the steps that are required to test the same?

Comment (chuyich) [reply docs/resources/security_certificate.md:22]: Please check the slack message.

Comment (chuyich) [reply docs/resources/security_certificate.md:22]: @csahu9 @wenjun666 has confirmed the fsx is supported.
Please add the following line:
* Amazon FSx for NetApp ONTAP

**Review by chuyich — CHANGES_REQUESTED** [review]
please update the document on both data source and resource with these:

## Supported Platforms
* On-perm ONTAP system 9.6 or higher
* Amazon FSx for NetApp ONTAP

**Comment by csahu9** [comment]
> please update the document on both data source and resource with these:
> 
> ## Supported Platforms
> * On-perm ONTAP system 9.6 or higher
> * Amazon FSx for NetApp ONTAP

@chuyich I've updated the resource docs. By the way is it on-perm or on-prem?
 PR for data source one is closed. Should I create a new issue for updating the docs for certificate data sources?
 Also, ACC tests build is failing even though I haven't made any changes in resource_test file. Could you please have a look?

File: docs/resources/security_certificate.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+### Related ONTAP commands
+```commandline
+* security certificate show
+* security certificate create
+* security certificate install
+* security certificate sign
+```
+
+## Supported Platforms
+* On-perm ONTAP system 9.6 or higher
```
Comment (chuyich) [inline docs/resources/security_certificate.md]: On-prem is the correct term. Pleas also update the data source(s), too.

**Review by carchi8py — APPROVED** [review]

**Review by suhasbshekar — APPROVED** [review]

**Review by chuyich — APPROVED** [review]
I have update all the docs. So approve it.

**Pull request merged**